llama_index/query_engine/multi_modal.py

```diff
@@ -92,6 +94,22 @@ def synthesize(
         additional_source_nodes: Optional[Sequence[NodeWithScore]] = None,
     ) -> RESPONSE_TYPE:
         image_nodes, text_nodes = _get_image_and_text_nodes(nodes)
+
+        # hack
```

Comment:
can you add a note on why and what it's doing?

Reply:
Thanks yeah, this was somewhat hacky business having to do with the case where we have an `ImageNode` but instead of using the default CLIP embedding, we want to generate a text description of the `ImageNode` and use that for its embedding representation. So, this outdated implementation was using a `TextNode` with an `image_path` in its metadata to get the job done.

Haotian have since discussed and we've determined a better route for now:
- User uses `ImageNode`
- if `text` field of `ImageNode` is populated, then it will compute `text_embedding` using the default/specified (text) `embed_model`.
- Moreover, this `text_embedding` will replace the `embedding` field so that indeed this text embedding can be used in downstream retrievals/queries.